### DISCUSSION TITLE
Using Weblate container for API tests

### DISCUSSION THREAD

**Discussion by gilfthde** [WeblateOrg/weblate#10980] [Q&A] [answered]
Hello, 

I'm currently working on a **Terraform provider** for Weblate to provision teams, projects, etc. This works fine so far. No I want to add tests for the Terraform resources. This usually works by spinning up a Docker container of the application (Weblate in this case) and then execute the tests against the API. The problem here is that I can only authenticate against the API with an access token and I haven't found a way yet to obtain one programmatically for the `admin` user on a freshly started Weblate container. 

The ugly workaround I'm using today is to use a database dump with a known token when starting Weblate with `docker compose`.

Any ideas on how to use Weblate in this scenario?

**Comment by nijel** [comment, marked as answer]
Not really elegant, but we use following in some automations:

```shell
weblate shell -c 'from weblate.auth.models import User; user = User.objects.create(username="monitor"); print(user.auth_token.key)'
```

**Reply by gilfthde** [reply]
Exactly what I needed. Many thanks.
